File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/utils/CoverHelper.java
// SPDX-License-Identifier: GPL-2.0-or-later

package org.dolphinemu.dolphinemu.utils;

import android.graphics.Bitmap;

import org.dolphinemu.dolphinemu.model.GameFile;

import java.io.FileOutputStream;

public final class CoverHelper
{
  public static String buildGameTDBUrl(GameFile game, String region)
  {
    String baseUrl = "https://art.gametdb.com/wii/cover/%s/%s.png";
    return String.format(baseUrl, region, game.getGameTdbId());
  }

  public static String getRegion(GameFile game)
  {
    String region;
    switch (game.getRegion())
    {
      case 0: // NTSC_J
        region = "JA";
        break;
      case 1: // NTSC_U
        region = "US";
        break;
      case 4: // NTSC_K
        region = "KO";
        break;
      case 2: // PAL
        switch (game.getCountry())
        {
          case 3: // Australia
            region = "AU";
            break;
          case 4: // France
            region = "FR";
            break;
          case 5: // Germany
            region = "DE";
            break;
          case 6: // Italy
            region = "IT";
            break;
          case 8: // Netherlands
            region = "NL";
            break;
          case 9: // Russia
            region = "RU";
            break;
          case 10: // Spain
            region = "ES";
            break;
          case 0: // Europe
          default:
            region = "EN";
            break;
        }
        break;
      case 3: // Unknown
      default:
        region = "EN";
        break;
    }
    return region;
  }

  public static void saveCover(Bitmap cover, String path)
  {
    try
    {
      FileOutputStream out = new FileOutputStream(path);
      cover.compress(Bitmap.CompressFormat.PNG, 100, out);
      out.close();
    }
    catch (Exception ignored)
    {
    }
  }
}

File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/utils/CoverHelper.kt
// SPDX-License-Identifier: GPL-2.0-or-later

package org.dolphinemu.dolphinemu.utils

import org.dolphinemu.dolphinemu.model.GameFile

object CoverHelper {
    @JvmStatic
    fun buildGameTDBUrl(game: GameFile, region: String?): String {
        val baseUrl = "https://art.gametdb.com/wii/cover/%s/%s.png"
        return String.format(baseUrl, region, game.gameTdbId)
    }

    @JvmStatic
    fun getRegion(game: GameFile): String {
        val region: String = when (game.region) {
            GameFile.REGION_NTSC_J -> "JA"
            GameFile.REGION_NTSC_U -> "US"
            GameFile.REGION_NTSC_K -> "KO"
            GameFile.REGION_PAL -> when (game.country) {
                3 -> "AU" // Australia
                4 -> "FR" // France
                5 -> "DE" // Germany
                6 -> "IT" // Italy
                8 -> "NL" // Netherlands
                9 -> "RU" // Russia
                10 -> "ES" // Spain
                0 -> "EN" // Europe
                else -> "EN"
            }
            3 -> "EN" // Unknown
            else -> "EN"
        }
        return region
    }
}
